include/ginkgo/core/base/executor.hpp

```diff
@@ -783,6 +783,39 @@ class ExecutorBase : public Executor {
 };
 
 
+/**
+ * Controls whether the DeviceReset function should be called thanks to a
+ * boolean.
+ */
+class EnableDeviceReset {
+public:
+    /**
+     * Toggles the Device Reset capability. If it was previously true, set
+     * false, and vice versa.
+     */
+    void toggle_device_reset() { device_reset_ = !device_reset_; }
```

Comment:
In my opinion, setting and getting are more common use cases than toggling and getting.